http-server/src/main/clojure/xtdb/jackson.clj

```diff
@@ -0,0 +1,72 @@
+(ns xtdb.jackson
+  (:require [xtdb.error :as err]
+            [jsonista.core :as json]
+            [jsonista.tagged :as jt])
+  (:import (clojure.lang Keyword PersistentHashSet)
+           (com.fasterxml.jackson.core JsonGenerator)
+           (com.fasterxml.jackson.databind ObjectMapper)
+           (com.fasterxml.jackson.databind.module SimpleModule)
+           (java.time Instant Duration LocalDate LocalDateTime ZonedDateTime)
+           (java.util Date Map)
+           (jsonista.jackson FunctionalSerializer)
+           (xtdb.jackson JsonLdValueOrPersistentHashMapDeserializer)))
+
+(defn serializer ^FunctionalSerializer [^String tag encoder]
+  (FunctionalSerializer.
+   (fn [value ^JsonGenerator gen]
+     (.writeStartObject gen)
+     (.writeStringField gen "@type" tag)
+     (.writeFieldName gen "@value")
+     (encoder value gen)
+     (.writeEndObject gen))))
+
+(defn encode-throwable [^Throwable t ^JsonGenerator gen]
+  (let [mapper ^ObjectMapper (.getCodec gen)]
+    (.writeStartObject gen)
+    (.writeStringField gen "xtdb.error/message" (.getMessage t))
+    (.writeStringField gen "xtdb.error/class" (.getName (.getClass t)))
+    (.writeFieldName gen "xtdb.error/data")
+    (.writeRawValue gen (.writeValueAsString mapper (ex-data t)))
```

Comment:
do either `writeObject` or `writePOJO` work here, rather than going down to `writeRawValue`?